src/other/utils.jl

```diff
@@ -178,3 +178,133 @@ function _nt_like_hash(v, h::UInt)
 
     return xor(objectid(Tuple(propertynames(v))), h)
 end
+
+_findall(B) = findall(B)
+
+@inline _blsr(x)= x & (x-1) 
+const _msk64 = ~UInt64(0)
+
+# findall optimized for B::BitVector
+# the main idea from Base.findall(B::BitArray)
+function _findall(B::BitVector)::Union{UnitRange{Int}, Vector{Int}}
+    nnzB = count(B)
+    nnzB == 0 && return Int[]
+    nnzB == length(B) && return 1:length(B)
+    I = Vector{Int}()
```

Comment:
you can also move it up to return `I` in line 191 for consistency.
```suggestion
    I = Vector{Int}(undef, nnzB)
```